README.md

```diff
@@ -36,5 +36,17 @@ Pull requests are welcome. For major changes, please open an issue first to disc
 
 Please make sure to update tests as appropriate.
 
+### Naming conventions
+
+We use [DDD (Domain Driven Design)](https://en.wikipedia.org/wiki/Domain-driven_design) principles.
+
+Files are sorted in folders that make a domain structure. Example: `flag/pixel/Pixel.ts` is located in the domain `pixel` which is a subdomain of the domain `flag`.
+Test files are located in a `spec` folder inside the domain folder they relate to. Example: all tests of stuff in the domain `flag` are located in `flag/spec/`.
+
+- Files defining a single class are named after the class (PascalCase). Example: `FlagRepository.ts` is expected to contain the definition of the `FlagRepository` class.
+- Other files are named after their content relative to their domain. Example: `flag-errors.ts` is expected to contain many error classes that are used in the `flag` domain.
```

Comment:
Two naming conventions can be ambiguous, maybe we should stick to PascalCase for everything?

Reply:
I do agree actually, but then because of how single-class files are named after the class, naming the errors.ts file something like FlagErrors.ts could make one think it defines a class named `FlagErrors`.

Should we just go for a "one file = one class" rule? I know this is pretty much java standard but it makes sense to me. I also believe that there is pretty much no cost to having a lot of very small files, and that it actually makes the project a lot more legible, especially with the domain-based folder structure.

Reply:
Fine with me